src/Datadog.Trace/Configuration/TracerSettings.cs

```diff
@@ -244,14 +245,24 @@ public static CompositeConfigurationSource CreateDefaultConfigurationSource()
 
         internal bool IsIntegrationEnabled(string name)
         {
-            bool disabled = Integrations[name].Enabled == false || DisabledIntegrationNames.Contains(name);
-            return TraceEnabled && !disabled;
+            if (TraceEnabled && !DomainMetadata.ShouldAvoidAppDomain())
```

Comment:
I think all 4 conditions can be expressed together, but I was outvoted some time ago and this was changed to use the `disabled` local variable. 😉 

```csharp
return TraceEnabled &&                            // tracing is enabled globally
       !DomainMetadata.ShouldAvoidAppDomain() &&  // this AppDomain looks good
	   Integrations[name].Enabled != false &&     // integration not explicitly disabled #1
	   !DisabledIntegrationNames.Contains(name);  // integration not explicitly disabled #2

```

Reply:
No need to change. Just adding some historical commentary.

Reply:
Sometimes readability rules!